tests/PersistentArrayTest.php

```diff
@@ -404,31 +419,188 @@ public function testLike()
+        $m->addField('country');
+        $m->addField('active', ['type' => 'boolean']);
+
+        // if no condition we should get all the data back
+        $iterator = $m->action('select');
+        $result = $m->persistence->applyScope($m, $iterator);
+        $this->assertInstanceOf(\atk4\data\Action\Iterator::class, $result);
+        $m->unload();
+        unset($iterator);
+        unset($result);
+
+        $m->scope()->clear();
+        $m->addCondition('country', 'REGEXP', 'Ireland|UK');
+        $result = $m->action('select')->get();
+        $this->assertSame(5, count($result));
+        $this->assertSame($a['countries'][1], $result[1]);
+        $this->assertSame($a['countries'][2], $result[2]);
+        $this->assertSame($a['countries'][4], $result[4]);
+        $this->assertSame($a['countries'][5], $result[5]);
+        $this->assertSame($a['countries'][6], $result[6]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('country', 'NOT REGEXP', 'Ireland|UK|Latvia');
+        $result = $m->action('select')->get();
+        $this->assertSame(1, count($result));
+        $this->assertSame($a['countries'][8], $result[8]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', '>', 18);
+        $result = $m->action('select')->get();
+        $this->assertSame(1, count($result));
+        $this->assertSame($a['countries'][9], $result[9]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', '>=', 18);
+        $result = $m->action('select')->get();
+        $this->assertSame(2, count($result));
+        $this->assertSame($a['countries'][8], $result[8]);
+        $this->assertSame($a['countries'][9], $result[9]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', '<', 12);
+        $result = $m->action('select')->get();
+        $this->assertSame(1, count($result));
+        $this->assertSame($a['countries'][1], $result[1]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', '<=', 12);
+        $result = $m->action('select')->get();
+        $this->assertSame(2, count($result));
+        $this->assertSame($a['countries'][1], $result[1]);
+        $this->assertSame($a['countries'][2], $result[2]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', [11, 12]);
+        $result = $m->action('select')->get();
+        $this->assertSame(2, count($result));
+        $this->assertSame($a['countries'][1], $result[1]);
+        $this->assertSame($a['countries'][2], $result[2]);
+        unset($result);
+        $m->unload();
+
+        $m->scope()->clear();
+        $m->addCondition('code', 'NOT IN', [11, 12, 13, 14, 15, 16, 17]);
```

Comment:
empty IN should be tested by DSQL, but add one test for it here - should return zero results, but not throw

needed because empty `IN()` is SQL emits an error normally

Reply:
done